### PR TITLE
41 add stat scaling system AND 40 add enemy scaling system AND 31 weapon upgrade improvements 

### PR DESCRIPTION
**Pull request by mitchellkazin15**
Created stat components for players, enemies, and weapons
Stats have a base value, as well as adders and multipliers that can be registered.
adders and multipliers modify the current stat with the eq (base + sum_adders) * sum_multipliers

Created scalers for player stats, enemy spawns and weapons.
When they level up new adders and multipliers are registered

### CODE REVIEW

File: autoloads/sound_manager.gd [hunk 1]
```diff
@@ -8,6 +8,7 @@ func _ready():
 
 
 func play_bgm(stream: AudioStream, volume_db: float = -10, force_restart: bool = false):
+	return
```
Comment (wchaney98) [inline autoloads/sound_manager.gd:11]: Down to keep these off for now but probably want a new issue for making a settings menu kind of thing?

**Review by wchaney98 — APPROVED** [review]
Gameplay comments:
* Weapons feel better now and seems to incentivize using all the different weapon types rather than just one
* Idea: a new smaller "trash" mob to give the player satisfaction of killing more things. Could make gameplay feel more rewarding
* Idea: render laser from sniper rifle barrel so the player can line up penetration shots (maybe an upgrade). Not super hard to do rn but would be a nice QOL thing

File: autoloads/sound_manager.gd (hunk 1 above)
Comment (mitchellkazin15) [reply autoloads/sound_manager.gd:11]: I'll put the sound back for now, I can work on a settings menu next

**Comment by mitchellkazin15** [comment]
> Gameplay comments:
> 
> * Weapons feel better now and seems to incentivize using all the different weapon types rather than just one
> * Idea: a new smaller "trash" mob to give the player satisfaction of killing more things. Could make gameplay feel more rewarding, less sitting and waiting for my guns to kill the current mobs
> * Idea: render laser from sniper rifle barrel so the player can line up penetration shots (maybe an upgrade). Not super hard to do rn but would be a nice QOL thing

These are good ideas, thanks! I'll make tickets for them

**Pull request merged**